tsdb/head_read.go

```diff
@@ -29,6 +30,99 @@ import (
 	"github.com/prometheus/prometheus/tsdb/index"
 )
 
+type postingsCacheEntry struct {
```

Comment:
Hm, are we sure we want to dynamically inject this and share within context. Is there more explicit alternative?

Reply:
designating it as an attribute of the IndexReader could be a better approach